### PR TITLE
Introducing new controller and model for MSIValidator 

### PR DESCRIPTION
**Pull request by vijaysaayi**
# MSI Validator
- Created a new controller and model for this functionality.
- To access this API, we need to make a Get request with the following query parameters:
   - **resource** : accepted values are keyvault, storage , sql - Mandatory parameter
   - **endpoint** : Mandatory to test the connectivity for keyvault and storage. For SQL this is taken from the connection string.
   - **clientId** : when specified , User Assigned Identity is used.

- Sample Inputs:
  - https://{webapp-name}.scm.azurewebsites.net/DaaS/api/msivalidator?resource=keyvault&endpoint={url_of_secret}
  - https://{webapp-name}.scm.azurewebsites.net/DaaS/api/msivalidator?resource=keyvault&clientId={client_id}&endpoint={url_of_secret}

  - https://{webapp-name}.scm.azurewebsites.net/DaaS/api/msivalidator?resource=storage&endpoint={url_of_blob}
  - https://{webapp-name}.scm.azurewebsites.net/DaaS/api/msivalidator?resource=storage&clientId={client_id}&endpoint={url_of_blob}

  - https://{webapp-name}.scm.azurewebsites.net/DaaS/api/msivalidator?resource=sql
  - https://{webapp-name}.scm.azurewebsites.net/DaaS/api/msivalidator?resource=sql&clientId={client_id}

### CODE REVIEW

File: DaaSRunner/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -38,5 +38,5 @@
 // [assembly: AssemblyVersion("1.0.*")]
 //[assembly: AssemblyVersion("1.0.0.0")]
 //[assembly: AssemblyFileVersion("1.0.0.0")]
-[assembly: AssemblyVersionAttribute("1.43.1120.01")]
-[assembly: AssemblyFileVersion("1.43.1120.01")]
+[assembly: AssemblyVersionAttribute("1.4.1120.01")]
```
[the remark this answers is not in the document]
Comment (vijaysaayi) [reply DaaSRunner/Properties/AssemblyInfo.cs]: Updated version to 1.44.1120.01

File: DiagnosticsExtension/Controllers/MsiValidatorController.cs [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+using System.Web.Http;
+
+namespace DiagnosticsExtension.Controllers
+{
+    [RoutePrefix("api/msivalidator")]
+    public class MsiValidatorController : ApiController
+    {
+        public class InputsForMsiValidator
+        {
+            public string clientId { get; set; }
```
[the remark this answers is not in the document]
Comment (vijaysaayi) [reply DiagnosticsExtension/Controllers/MsiValidatorController.cs]: Refactored code following c# naming conventions

File: DiagnosticsExtension/Models/MsiValidatorModels.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+
+namespace DiagnosticsExtension.Models
+{
+    public class AccessTokenInformation
+    {
+        [JsonProperty("access_token")]
+        public string accessToken;
```
[the remark this answers is not in the document]
Comment (vijaysaayi) [reply DiagnosticsExtension/Models/MsiValidatorModels.cs]: Updated property name to AccessToken. The response from MSI service has key name as access_token. Hence using JsonProperty to help in deserialization.

File: DiagnosticsExtension/Controllers/MsiValidatorController.cs [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+                }
+                else
+                {
+                    Result.GetTokenTestResult.ErrorDetails = JsonConvert.DeserializeObject<AdalError>(await response.Content.ReadAsStringAsync());
+                }
+
+                return response.IsSuccessStatusCode;
+            }
+
+            public async Task TestConnectivityAsync(MsiValidatorInput input)
```
Comment (puneetg1983) [inline DiagnosticsExtension/Controllers/MsiValidatorController.cs:196]: should this be public?

**Review by puneetg1983 — CHANGES_REQUESTED** [review]

**Review by puneetg1983 — APPROVED** [review]
:shipit:

**Pull request merged**